main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -292,4 +292,29 @@ object TypeError {
       vt << Underline("Tip:") << " Add an instance for the type." << NewLine
     }
   }
+
+  /**
+    * An error indicating that the main function's scheme is incorrect.
+    *
+    * @param actualScheme   the erroneous function's scheme.
+    * @param expectedScheme the scheme the main function is expected to have.
+    * @param loc            the location where the error occurred.
+    */
+  case class IllegalMain(actualScheme: Scheme, expectedScheme: Scheme, loc: SourceLocation) extends TypeError {
+    override def summary: String = "Illegal main."
+
+    def message: VirtualTerminal = {
+      val vt = new VirtualTerminal()
+      vt << Line(kind, source.format) << NewLine
+      vt << ">> Main function with wrong type." << NewLine
+      vt << NewLine
+      vt << Code(loc, s"main function with wrong type.") << NewLine
+      vt << NewLine
+      vt << s"  Expected: " << Cyan(FormatScheme.formatScheme(expectedScheme)) << NewLine
+      vt << s"  Actual:   " << Magenta(FormatScheme.formatScheme(actualScheme)) << NewLine
+      vt << NewLine
```

Comment:
While the above is appealing for its regularity, it might be better if we simply wrote a longer message ala:

```
In Flix the main function must have the form:

  def main(args: Array[String]): Int32 & Impure = ...

i.e. 
- it must return an integer which is the exit code, and
- it must have a side-effect (such as printing to the screen).

(If the arguments are not needed args can be replaced by _.)

```

or something to that effect.